File: Backend/Events/Converters/Parties/DigievolutionConverter.cs
using Backend.Domain.Models.Parties.Digimons;
using Backend.Events.DTO.Party.Digimon;

namespace Backend.Events.Converters.Parties;

public static class DigievolutionConverter
{
    public static DigievolutionDTO ToDTO(Digievolution digievolution) => new()
    {
        Level = digievolution.Level
    };
}

File: Frontend/src/events/converters/parties/digimons/digievolution.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { DigievolutionDTO } from "@/events/dto/parties/digimons/digievolution.dto";
import type { Digievolution } from "@/models";

export class DigievolutionConverter {
  public static convert(digievolutionDto: DeepRequired<DigievolutionDTO>): Digievolution {
    return {
      level: digievolutionDto.level,
      dvexp: digievolutionDto.dvexp,
    };
  }
}
